src/solve.rs

```diff
@@ -279,12 +295,16 @@ fn is_market_order(tokens: &BTreeMap<H160, TokenInfoModel>, order: OrderModel) -
 
 async fn get_swaps_for_orders_from_zeroex(
     orders: Vec<(usize, OrderModel)>,
+    slippage_context: &SlippageContext<'_>,
     api_key: Option<String>,
 ) -> Result<Vec<((usize, OrderModel), (SwapQuery, SwapResponse))>> {
     let zeroex_futures = orders
         .into_iter()
         .filter(|(_, x)| !x.is_liquidity_order)
         .map(|(index, order)| {
+            let slippage = slippage_context
+                .relative_for_order(&order)
+                .unwrap_or(FALLBACK_SLIPPAGE);
```

Comment:
One downside here is that it runs into a similar problem as the Baseline and Quasimodo solver and that you don't necessarily have external prices for all tokens in the trade route.

Its good that there is a fallback (and a low-effort change that we should do in the services repo as well). Ideally, it fallback to using clearing prices in case external prices aren't available - but that is a much more complicated change.